test/stablex/batch_exchange.js

```diff
@@ -2044,14 +2044,34 @@ contract("BatchExchange", async accounts => {
   })
   describe("Large Examples", () => {
     it("ensures hard gas limit on largest possible ring trade ", async () => {
-      const batchExchange = await setupGenericStableX(25)
-      const sixPointFiveMillion = 6500000
-      await makeDeposits(batchExchange, accounts, longRingTrade.deposits)
+      const batchExchange = await setupGenericStableX(30)
+      const fivePointFiveMillion = 5500000
+
+      const tradeExample = largeRing30
+
+      // Deposit double sufficient amount and immediately request withdraw for half
+      await makeDeposits(batchExchange, accounts, tradeExample.deposits, 2)
+      for (const deposit of tradeExample.deposits) {
+        const tokenAddress = await batchExchange.tokenIdToAddressMap.call(deposit.token)
+        await batchExchange.requestWithdraw(tokenAddress, deposit.amount, { from: accounts[deposit.user] })
+      }
+      await closeAuction(batchExchange)
+
       const batchId = (await batchExchange.getCurrentBatchId.call()).toNumber()
-      const orderIds = await placeOrders(batchExchange, accounts, longRingTrade.orders, batchId + 1)
+      const orderIds = await placeOrders(batchExchange, accounts, tradeExample.orders, batchId + 1)
       await closeAuction(batchExchange)
 
-      const solution = solutionSubmissionParams(longRingTrade.solutions[0], accounts, orderIds)
+      // Ensure that first 30 orders have valid withdraw requests.
+      for (const deposit of tradeExample.deposits.slice(0, 30)) {
+        const tokenAddress = await batchExchange.tokenIdToAddressMap.call(deposit.token)
+        assert(
```

Comment:
How long is this test taking? I'm wondering if we should make all these calls here or if one check is sufficient?

Reply:
```
    Large Examples
      ✓ ensures hard gas limit on largest possible ring trade  (35710ms)
```

I could reduce this to one assertion.

Reply:
35 seconds for a single test is really long. Maybe we can at least not sequentially await on all future but instead issue the requests in parallel? That is, use [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) instead of sequential awaits.

Reply:
I like this idea much better!